File: sort_test.py
import random
import time
import sys

def generate_random_array(size):
    return [random.randint(0, 1000000) for _ in range(size)]

def run_load_test(n):
    print(f"Starting sort test with array size: {n}")
    large_array = generate_random_array(n)
    
    print("Sorting array...")
    start_time = time.time()
    
    large_array.sort()  # Sort the array in place
    
    end_time = time.time()
    print(f"Sorting completed in {end_time - start_time:.2f} seconds.")

def cleanup():
    print("Performing cleanup...")

def main():
    if len(sys.argv) < 2:
        print("Error: Please specify a mode ('run' or 'cleanup').")
        return

    mode = sys.argv[1]
    
    if mode == 'run':
        if '-n' in sys.argv:
            try:
                n_index = sys.argv.index('-n') + 1
                n = int(sys.argv[n_index])
                run_load_test(n)
            except (IndexError, ValueError):
                print("Error: Please supply a valid number after '-n'")
        else:
            print("Error: Please supply the '-n' argument.")
    elif mode == 'cleanup':
        cleanup()
    else:
        print(f"Error: Unknown mode '{mode}'")

if __name__ == "__main__":
    main()

File: SortTest.java
import java.util.Arrays;
import java.util.Random;

public class SortTest {

    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("Error: Please specify a mode ('run' or 'cleanup').");
            return;
        }

        String mode = args[0];

        if ("run".equals(mode)) {
            runLoadTest(args);
        } else if ("cleanup".equals(mode)) {
            cleanup();
        } else {
            System.out.println("Error: Unknown mode " + mode);
        }
    }

    private static void runLoadTest(String[] args) {
        System.out.println("Starting sort test");

        int n = 0;
        for (int i = 1; i < args.length; i++) {
            if ("-n".equals(args[i]) && i + 1 < args.length) {
                n = Integer.parseInt(args[i + 1]);
            }
        }

        if (n > 0) {
            System.out.println("Generating array of size: " + n);
            int[] largeArray = generateRandomArray(n);
            System.out.println("Sorting array...");
            long startTime = System.currentTimeMillis();
            
            Arrays.sort(largeArray);

            long endTime = System.currentTimeMillis();
            System.out.println("Sorting completed in " + (endTime - startTime) + " ms");
        } else {
            System.out.println("Please supply -n argument");
        }
    }

    private static int[] generateRandomArray(int size) {
        Random random = new Random();
        int[] array = new int[size];
        for (int i = 0; i < size; i++) {
            array[i] = random.nextInt();
        }
        return array;
    }

    private static void cleanup() {
        System.out.println("Performing cleanup...");
    }
}
